scripts/dev/e2e.py

```diff
@@ -140,9 +143,35 @@ def create_test_runner_pod(test: str):
     dev_config = load_config()
     corev1 = client.CoreV1Api()
     pod_body = _get_testrunner_pod_body(test)
+
+    if not wait_for_condition(
+        lambda: corev1.list_namespaced_pod(
+            dev_config.namespace, field_selector="metadata.name==" + TEST_RUNNER_NAME
```

Comment:
[nit] we could use string.format or f-strings

`"metadata.name=={}".format(TEST_RUNNER_NAME)`

`f"metadata.name=={TEST_RUNNER_NAME)"`